app/src/main/java/org/mozilla/focus/locale/screen/LanguageFragment.kt

```diff
@@ -5,71 +5,68 @@
 import mozilla.components.lib.state.ext.observeAsComposableState
 import mozilla.components.support.locale.LocaleUseCases
 import org.mozilla.focus.R
 import org.mozilla.focus.ext.components
 import org.mozilla.focus.ext.requireComponents
-import org.mozilla.focus.settings.BaseSettingsComposeFragment
+import org.mozilla.focus.settings.BaseComposeFragment
 import org.mozilla.focus.state.AppAction
 import org.mozilla.focus.state.Screen
 
-class LanguageFragment : BaseSettingsComposeFragment() {
-    private val browserStore by lazy { requireContext().components.store }
-    private val localeUseCase by lazy { LocaleUseCases(browserStore) }
-    private val languageScreenStore by lazy {
-        LanguageScreenStore(
+class LanguageFragment : BaseComposeFragment() {
+
+    private lateinit var browserStore: BrowserStore
+    private lateinit var localeUseCase: LocaleUseCases
+    private lateinit var languageScreenStore: LanguageScreenStore
+    private lateinit var screenInteractorDefault: DefaultLanguageScreenInteractor
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        browserStore = requireContext().components.store
+        localeUseCase = LocaleUseCases(browserStore)
+        languageScreenStore = LanguageScreenStore(
             LanguageScreenState(),
             listOf(
                 LanguageMiddleware(
                     activity = requireActivity(), localeUseCase = localeUseCase
                 )
             )
         )
-    }
-    private val screenInteractorDefault by lazy {
-        DefaultLanguageScreenInteractor(
+        screenInteractorDefault = DefaultLanguageScreenInteractor(
             languageScreenStore = languageScreenStore
         )
     }
 
-    override fun onCreateView(
-        inflater: LayoutInflater,
-        container: ViewGroup?,
-        savedInstanceState: Bundle?
-    ): View {
-        return ComposeView(requireContext()).apply {
-            setContent {
-                toolbar(
-                    title = getString(R.string.preference_language),
-                    onClickUpButton = {
-                        requireComponents.appStore.dispatch(
-                            AppAction.OpenSettings(Screen.Settings.Page.General)
-                        )
-                    }
-                ) {
-                    val languagesList = languageScreenStore
-                        .observeAsComposableState { state -> state.languageList }.value
-                    val languageSelected = languageScreenStore.observeAsComposableState { state ->
-                        state.selectedLanguage
-                    }.value
-                    if (languageSelected != null && languagesList != null) {
-                        Languages(languageSelected = languageSelected, languages = languagesList)
-                    }
-                }
-            }
+    override val title: String
```

Comment:
nit: This could be a `stringRes` and it would make the getter obsolete, but I am fine with either.

Reply:
Passing just the Int for `titleRes` sounds great!